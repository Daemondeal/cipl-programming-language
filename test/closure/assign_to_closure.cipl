let f
let g

    let local = "local"
    proc f_():
        print(local)
        local = "after f"
        print(local)

    f = f_

    proc g_():
        print(local)
        local = "after g"
        print(local)

    g = g_

f()
g()