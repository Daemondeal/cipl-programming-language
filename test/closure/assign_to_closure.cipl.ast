(let f null)
(let g null)
(block
    (let local local)
    (proc f_ (body
        (expression (call println local))
        (expression (assign local after f))
        (expression (call println local))
    ))
    (expression (assign f f_))
    (proc g_ (body
        (expression (call println local))
        (expression (assign local after g))
        (expression (call println local))
    ))
    (expression (assign g g_))
)
(expression (call f))
(expression (call g))

