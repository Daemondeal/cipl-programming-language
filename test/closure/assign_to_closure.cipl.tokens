Token          Lexeme         Line 
LET            let            1    
IDENTIFIER     f              1    
LINE_END       \n             2    
LET            let            2    
IDENTIFIER     g              2    
LINE_END       \n             3    
INDENT                        4    
LET            let            4    
IDENTIFIER     local          4    
EQUAL          =              4    
STRING         "local"        4     local
LINE_END       \n             5    
PROC           proc           5    
IDENTIFIER     f_             5    
LEFT_PAREN     (              5    
RIGHT_PAREN    )              5    
COLON          :              5    
LINE_END       \n             6    
INDENT                        6    
IDENTIFIER     println        6    
LEFT_PAREN     (              6    
IDENTIFIER     local          6    
RIGHT_PAREN    )              6    
LINE_END       \n             7    
IDENTIFIER     local          7    
EQUAL          =              7    
STRING         "after f"      7     after f
LINE_END       \n             8    
IDENTIFIER     println        8    
LEFT_PAREN     (              8    
IDENTIFIER     local          8    
RIGHT_PAREN    )              8    
LINE_END       \n             9    
DEDENT                        10   
IDENTIFIER     f              10   
EQUAL          =              10   
IDENTIFIER     f_             10   
LINE_END       \n             11   
PROC           proc           12   
IDENTIFIER     g_             12   
LEFT_PAREN     (              12   
RIGHT_PAREN    )              12   
COLON          :              12   
LINE_END       \n             13   
INDENT                        13   
IDENTIFIER     println        13   
LEFT_PAREN     (              13   
IDENTIFIER     local          13   
RIGHT_PAREN    )              13   
LINE_END       \n             14   
IDENTIFIER     local          14   
EQUAL          =              14   
STRING         "after g"      14    after g
LINE_END       \n             15   
IDENTIFIER     println        15   
LEFT_PAREN     (              15   
IDENTIFIER     local          15   
RIGHT_PAREN    )              15   
LINE_END       \n             16   
DEDENT                        17   
IDENTIFIER     g              17   
EQUAL          =              17   
IDENTIFIER     g_             17   
LINE_END       \n             18   
DEDENT                        19   
IDENTIFIER     f              19   
LEFT_PAREN     (              19   
RIGHT_PAREN    )              19   
LINE_END       \n             20   
IDENTIFIER     g              20   
LEFT_PAREN     (              20   
RIGHT_PAREN    )              20   
LINE_END       \n             20   
EOF                           20   
