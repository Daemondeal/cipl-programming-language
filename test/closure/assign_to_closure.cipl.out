::stdout::
local
after f
after f
after g

::stderr::
